tests/test_cases.rs

```diff
@@ -76,11 +76,13 @@ fn test_generation(name: &str) -> Result<Vec<config::Device>> {
 }
 
 #[test]
+#[ignore] // Needs root for the modprobe call
 fn test_01_basic() {
     test_generation("01-basic").unwrap();
 }
 
 #[test]
+#[ignore] // Needs root for the modprobe call
```

Comment:
Instead of requiring manual manipulation of which tests are run, it'd be much better to simply exit if we lack privileges. With a growing number of tests, it'd soon become very cumbersome to enable and disable tests by hand like this.

Reply:
There's no way to mark a test skipped from inside said test, so the test can either fail (which, well, isn't really true) or succeed (which is an even worse "the tests succeed but don't actually test anything and are entirely useless" false positive). This is the only way to ensure vaguely correct semantics.

Reply:
Updated to only run modprobe if the module doesn't exist, so doesn't require root.